src/pages/iou/IOUModal.js

```diff
@@ -41,35 +58,54 @@ class IOUModal extends Component {
         this.navigateToNextStep = this.navigateToNextStep.bind(this);
         this.updateAmount = this.updateAmount.bind(this);
         this.currencySelected = this.currencySelected.bind(this);
-
+        this.createTransaction = this.createTransaction.bind(this);
+        this.updateComment = this.updateComment.bind(this);
         this.addParticipants = this.addParticipants.bind(this);
+
         this.state = {
             currentStepIndex: 0,
             participants: [],
+
+            // amount is currency in decimal format
             amount: '',
             selectedCurrency: 'USD',
             isAmountPageNextButtonDisabled: true,
+            comment: '',
         };
     }
 
     componentDidMount() {
         getPreferredCurrency();
     }
 
+    componentDidUpdate(prevProps) {
+        // Successfully close the modal if transaction creation has ended and theree is no error
+        if (prevProps.iou.creatingIOUTransaction && !this.props.iou.creatingIOUTransaction && !this.props.iou.error) {
+            Navigation.dismissModal();
+        }
+    }
+
     /**
      * Retrieve title for current step, based upon current step and type of IOU
      *
      * @returns {String}
      */
 
     getTitleForStep() {
-        if (this.state.currentStepIndex === 1) {
+        const currentStepIndex = this.state.currentStepIndex;
+        if (currentStepIndex === 1 || currentStepIndex === 2) {
             return `${this.props.hasMultipleParticipants ? 'Split' : 'Request'} $${this.state.amount}`;
         }
-        if (steps[this.state.currentStepIndex] === Steps.IOUAmount) {
+        if (steps[currentStepIndex] === Steps.IOUAmount) {
```

Comment:
we check the index in line 96, but the step name in line 99. Lets be consistent and just use `currentStepIndex` === 0